components/server/src/prebuilds/github-app.ts

```diff
@@ -112,6 +114,19 @@ export class GithubApp {
                     res.redirect(301, this.getBadgeImageURL());
                 });
 
+        // Use RequestContext
+        options.getRouter &&
+            options.getRouter().use((req, res, next) => {
+                runWithRequestContext(
```

Comment:
not sure how to test it, but code wise it looks ok

Reply:
Tested that we still receive events: Preview environments are configured to use the GitHub App for that:
![image](https://github.com/gitpod-io/gitpod/assets/32448529/ad6d44b3-a598-43f5-9d91-18a4c9557e4f)